fpsd/evaluation.py

```diff
@@ -0,0 +1,193 @@
+import numpy as np
+from scipy import interp
+from sklearn import metrics
+import pandas as pd
+import pdb
+
+import matplotlib as mpl
+mpl.use("Agg")
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+THRESHOLDS = [0.01, 0.05, 0.1, 0.5, 1, 5, 10]
+
+def get_metrics(truth, predicted):
+    """Compute evaluation metrics for a given fold
+
+    Args:
+        truth [list]: List of labels
+        predicted [list]: List of predicted scores
+
+    Returns:
+        eval_metrics [dict]: dict of metrics to be saved in the db
+    """
+
+    eval_metrics = {}
+    fpr, tpr, thresholds = metrics.roc_curve(truth, predicted)
+    eval_metrics["fpr"] = fpr
+    eval_metrics["tpr"] = tpr
+    eval_metrics["thresholds"] = thresholds
+    eval_metrics["auc"] = metrics.auc(fpr, tpr)
+    eval_metrics["precision"] = {}
+    eval_metrics["recall"] = {}
+    eval_metrics["f1"] = {}
+
+    for threshold in THRESHOLDS:
+        precision, recall, f1 = precision_recall_at_x_proportion(truth,
+            predicted, x_proportion=threshold/100.)
+        eval_metrics.update({threshold: {'precision': precision, 
+                                         'recall': recall,
+                                         'f1': f1}})
+    return eval_metrics
+
+
+def get_average_metrics(metrics_list):
```

Comment:
Might need to change a variable name... would much of the last suite starting at line 63 I believe.